packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx

```diff
@@ -0,0 +1,343 @@
+export interface TipsEntranceDialogProps {
+    open: boolean
+    onClose: () => void
+}
+const useStyles = makeStyles()((theme) => ({
+    walletBtn: {
+        fontSize: '14px',
+    },
+    alertBox: {
+        marginBottom: '20px',
+    },
+    actions: {
+        position: 'absolute',
+        bottom: 16,
+        width: 'calc( 100% - 32px)',
+        display: 'flex',
+        alignItems: 'center',
+        justifyContent: 'center',
+        gap: theme.spacing(1.5),
+    },
+    dContent: {
+        height: 600,
+        position: 'relative',
+        boxSizing: 'border-box',
+    },
+    btnContainer: {
+        width: '100%',
+        display: 'flex',
+        flexDirection: 'row-reverse',
+    },
+    loading: {
+        position: 'absolute',
+        top: '50%',
+        left: '50%',
+        transform: 'translate(-50%,-50%)',
+    },
+    cancelBtn: {
+        border: '1px solid rgba(47, 51, 54, 1)',
+    },
+}))
+
+enum BodyViewSteps {
+    main = 'Tips',
+    setting = 'Settings',
+    wallets = 'Wallets',
+    addWallet = 'Add wallet',
+}
+export interface WalletProof extends BindingProof {
+    isDefault: number
+    isPublic: number
+}
+export function TipsEntranceDialog({ open, onClose }: TipsEntranceDialogProps) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const [showAlert, setShowAlert] = useState(true)
+    const supportedNetworks = TipsSupportedChains
+    const [bodyView, setBodyView] = useState<BodyViewSteps>(BodyViewSteps.main)
+    const [hasChanged, setHasChanged] = useState(false)
+    const [rawPatchData, setRawPatchData] = useState<WalletProof[]>([])
+    const [rawWalletList, setRawWalletList] = useState<WalletProof[]>([])
+    const { showSnackbar } = useCustomSnackbar()
+    const nowTime = formatDateTime(new Date(), 'yyyy-MM-dd HH:mm')
+
+    const { value: currentPersonaIdentifier } = useAsyncRetry(
+        () => Services.Settings.getCurrentPersonaIdentifier(),
+        [open],
+    )
+    const { value: currentPersona } = useAsyncRetry(
+        () => Services.Identity.queryPersona(currentPersonaIdentifier as ECKeyIdentifier),
+        [currentPersonaIdentifier],
+    )
+    const clickBack = () => {
+        if (bodyView === BodyViewSteps.main) {
+            onClose()
+        } else {
+            setBodyView(BodyViewSteps.main)
+        }
+    }
+    const { value: kv, retry: retryKv } = useKvGet()
+    const { loading, value: proofRes, retry: retryProof } = useProvedWallets()
+    useEffect(() => {
+        setShowAlert(true)
+    }, [open])
+    useEffect(() => {
+        setHasChanged(false)
+        const walletsList = proofRes
+            ? (proofRes as NextIDPersonaBindings).proofs.filter((x) => x.platform === NextIDPlatform.Ethereum)
+            : []
+        walletsList.sort((a, b) => Number.parseInt(a.created_at, 10) - Number.parseInt(b.created_at, 10))
+        if (kv !== undefined && (kv?.val as NextIdStorageInfo).proofs.length > 0 && walletsList.length > 0) {
```

Comment:
Check `kv.ok` before using `kv.val`, if `kv.ok` is true, then `kv.val` would be the desired type.

```suggestion
        if (kv?.ok && kv.val.proofs.length > 0 && walletsList.length > 0) {
```

Reply:
```diff
diff --git a/packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx b/packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx
index d3909ee15..7ccf2f999 100644
--- a/packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx
+++ b/packages/mask/src/plugins/Tips/SNSAdaptor/TipsEntranceDialog.tsx
@@ -107,7 +107,7 @@ export function TipsEntranceDialog({ open, onClose }: TipsEntranceDialogProps) {
             setBodyView(BodyViewSteps.main)
         }
     }
-    const { value: kv, retry: retryKv } = useKvGet()
+    const { value: kv, retry: retryKv } = useKvGet<NextIdStorageInfo>()
     const { loading, value: proofRes, retry: retryProof } = useProvedWallets()
     useEffect(() => {
         setShowAlert(true)
@@ -118,7 +118,7 @@ export function TipsEntranceDialog({ open, onClose }: TipsEntranceDialogProps) {
             ? (proofRes as NextIDPersonaBindings).proofs.filter((x) => x.platform === NextIDPlatform.Ethereum)
             : []
         walletsList.sort((a, b) => Number.parseInt(a.created_at, 10) - Number.parseInt(b.created_at, 10))
-        if (kv !== undefined && (kv?.val as NextIdStorageInfo).proofs.length > 0 && walletsList.length > 0) {
+        if (kv?.ok && kv.val.proofs.length > 0 && walletsList.length > 0) {
             const kvCache = (kv.val as NextIdStorageInfo).proofs.find(
                 (x) => x.identity === currentPersona?.publicHexKey,
             )
diff --git a/packages/mask/src/plugins/Tips/hooks/useKv.ts b/packages/mask/src/plugins/Tips/hooks/useKv.ts
index f04d942f6..bf61e6f0a 100644
--- a/packages/mask/src/plugins/Tips/hooks/useKv.ts
+++ b/packages/mask/src/plugins/Tips/hooks/useKv.ts
@@ -13,10 +13,10 @@ const getCurrentPersonaPublicKey = async () => {
     if (!currentPersona || !currentPersona.publicHexKey) return ''
     return currentPersona.publicHexKey
 }
-export function useKvGet() {
+export function useKvGet<T>() {
     return useAsyncRetry(async () => {
         const publicHexKey = await getCurrentPersonaPublicKey()
-        return NextIDStorage.get(publicHexKey)
+        return NextIDStorage.get<T>(publicHexKey)
     })
 }
 
diff --git a/packages/web3-providers/src/NextID/kv.ts b/packages/web3-providers/src/NextID/kv.ts
index 4f094e7a6..58171da7e 100644
--- a/packages/web3-providers/src/NextID/kv.ts
+++ b/packages/web3-providers/src/NextID/kv.ts
@@ -2,7 +2,7 @@
  * Document url: https://github.com/nextdotid/kv_server/blob/develop/docs/api.apib
  */
 import urlcat from 'urlcat'
-import type { NextIDStoragePayload, NextIDPlatform, NextIdStorageInfo } from '@masknet/shared-base'
+import type { NextIDStoragePayload, NextIDPlatform } from '@masknet/shared-base'
 import { fetchJSON } from './helper'
 import type { Result } from 'ts-results'
 import type { NextIDBaseAPI } from '../types'
@@ -29,7 +29,7 @@ export class NextIDStorageAPI implements NextIDBaseAPI.Storage {
      * @param personaPublicKey
      *
      */
-    async get<T>(personaPublicKey: string): Promise<Result<NextIdStorageInfo, string>> {
+    async get<T>(personaPublicKey: string): Promise<Result<T, string>> {
         const full = await fetchJSON<{ [MASK_STORAGE_KEY]: T }>(
             urlcat(BASE_URL, '/v1/kv', { persona: personaPublicKey }),
         )

```